### PR TITLE
TF: add Shared-VPC jobs

### PR DESCRIPTION
**Pull request by yunjiang29**

### CODE REVIEW

File: ci-operator/config/terraform-redhat/terraform-provider-rhcs/terraform-redhat-terraform-provider-rhcs-main.yaml [hunk 1]
```diff
@@ -62,6 +62,22 @@ tests:
...
     - ref: rhcs-e2e-tests
     workflow: rhcs-aws-sts
+- as: rosa-sts-shared-vpc-critical-high-f3
+  cron: 22 15 1,4,7,10,13,16,19,22,25,28 * *
+  steps:
+    cluster_profile: aws-terraform-qe
+    env:
+      CASE_LABEL_FILTER: (Critical,High)&&(day1-post,day2)&&!Exclude
+      CHANNEL_GROUP: candidate
+      CLUSTER_PROFILE: rosa-sts-sv
```
Comment (xueli181114) [inline ci-operator/config/terraform-redhat/terraform-provider-rhcs/terraform-redhat-terraform-provider-rhcs-main.yaml:72]: I didn't see this cluster profile in the profiles file. Is it merged?

File: ci-operator/config/terraform-redhat/terraform-provider-rhcs/terraform-redhat-terraform-provider-rhcs-main.yaml [hunk 2]
```diff
@@ -62,6 +62,22 @@ tests:
...
+      ENABLE_SHARED_VPC: "yes"
+      QE_USAGE: rosa-sts-shared-vpc-critical-high-f3
+      REGION: us-east-1
```
Comment (xueli181114) [inline ci-operator/config/terraform-redhat/terraform-provider-rhcs/terraform-redhat-terraform-provider-rhcs-main.yaml:75]: Is there any function enabled in the profile those cannot fit for current region?

File: ci-operator/step-registry/rhcs/e2e/setup/rhcs-e2e-setup-ref.yaml [hunk 3]
```diff
@@ -29,6 +29,8 @@ ref:
     default: ""
   - name: "RHCS_VERSION" #if setup a job with RHCS_SOURCE the source in all tf manifest will be updated
     default: ""
+  - name: "ENABLE_SHARED_VPC"
```
Comment (xueli181114) [inline ci-operator/step-registry/rhcs/e2e/setup/rhcs-e2e-setup-ref.yaml:32]: Will this env variable overwrite the configuration in profile?

File: ci-operator/step-registry/rhcs/e2e/teardown/rhcs-e2e-teardown-ref.yaml [hunk 4]
```diff
@@ -25,6 +25,8 @@ ref:
     default: ""
   - name: "RHCS_VERSION" #if setup a job with RHCS_SOURCE the source in all tf manifest will be updated
     default: ""
+  - name: "ENABLE_SHARED_VPC"
```
Comment (xueli181114) [inline ci-operator/step-registry/rhcs/e2e/teardown/rhcs-e2e-teardown-ref.yaml:28]: Is tear down part covered the share vpc clean up?

File: ci-operator/config/terraform-redhat/terraform-provider-rhcs/terraform-redhat-terraform-provider-rhcs-main.yaml (hunk 1 above)
Comment (yunjiang29) [reply ci-operator/config/terraform-redhat/terraform-provider-rhcs/terraform-redhat-terraform-provider-rhcs-main.yaml:72]: It's under reviewing, https://github.com/terraform-redhat/terraform-provider-rhcs/pull/569/files#diff-1afc179e83a78ce9bcfc4e2868861c93c2f089d81345e0c7a0d4944d80d93a40R67

File: ci-operator/config/terraform-redhat/terraform-provider-rhcs/terraform-redhat-terraform-provider-rhcs-main.yaml (hunk 2 above)
Comment (yunjiang29) [reply ci-operator/config/terraform-redhat/terraform-provider-rhcs/terraform-redhat-terraform-provider-rhcs-main.yaml:75]: No.
Actually, `us-east-1` is more complicated  than other regions, see https://github.com/terraform-redhat/terraform-provider-rhcs/pull/569/files#diff-af47971dcdc8b7bd6bff3593ed96629cb6ce60b778725013948b3e9c5e133d0aR506-R510

File: ci-operator/step-registry/rhcs/e2e/setup/rhcs-e2e-setup-ref.yaml (hunk 3 above)
Comment (yunjiang29) [reply ci-operator/step-registry/rhcs/e2e/setup/rhcs-e2e-setup-ref.yaml:32]: No. It's for Prow job only. 

The configuration for profile is `shared_vpc: true` https://github.com/terraform-redhat/terraform-provider-rhcs/pull/569/files#diff-1afc179e83a78ce9bcfc4e2868861c93c2f089d81345e0c7a0d4944d80d93a40R100

File: ci-operator/step-registry/rhcs/e2e/teardown/rhcs-e2e-teardown-ref.yaml (hunk 4 above)
Comment (yunjiang29) [reply ci-operator/step-registry/rhcs/e2e/teardown/rhcs-e2e-teardown-ref.yaml:28]: Yes, see changes in `DestroyRHCSClusterByProfile` https://github.com/terraform-redhat/terraform-provider-rhcs/pull/569/files#diff-af47971dcdc8b7bd6bff3593ed96629cb6ce60b778725013948b3e9c5e133d0aR664-R712

File: ci-operator/step-registry/rhcs/e2e/setup/rhcs-e2e-setup-commands.sh [hunk 5]
```diff
@@ -34,6 +34,11 @@ if [ -z "${RHCS_TOKEN}" ]; then
 fi
 export RHCS_TOKEN=${RHCS_TOKEN}
 export AWS_SHARED_CREDENTIALS_FILE=${CLUSTER_PROFILE_DIR}/.awscred
+
+if [[ ${ENABLE_SHARED_VPC} == "yes" ]]; then
```
Comment (radtriste) [inline ci-operator/step-registry/rhcs/e2e/setup/rhcs-e2e-setup-commands.sh:38]: why not using `true|false` ?

Comment (yunjiang29) [reply ci-operator/step-registry/rhcs/e2e/setup/rhcs-e2e-setup-commands.sh:38]: `ENABLE_SHARED_VPC` is being used by OCP as well, to keep the parameter consistent in this repo, `yes|no` is used.

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-shared-vpc-critical-high-f3

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-shared-vpc-critical-high-f3

**Comment by yunjiang29** [comment]
@xueli181114 some e2e cases failed, can you help check with them?

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-shared-vpc-critical-high-f3

**Comment by xueli181114** [comment]
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-shared-vpc-critical-high-f3

**Comment by xueli181114** [comment]
/pj-rehearse pull-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-advanced-critical-high-presubmit

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-shared-vpc-critical-high-f3

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-shared-vpc-critical-high-f3

**Comment by yunjiang29** [comment]
@nirarg can you review and approve this PR? Thanks

**Comment by nirarg** [comment]
@yunjiang29 the rehearsal is failing
Can you please describe why and if it can be merge with the error?

**Pull request merged**